web/client/epics/featuregrid.js

```diff
@@ -566,8 +630,25 @@ module.exports = {
             })
             .switchMap(() => {
                 const {query: q, featuregrid: f} = store.getState();
+                const layerId = (f || {}).selectedLayer;
                 const filter = (q || {}).filterObj;
-                const layer = (f || {}).selectedLayer;
-                return Rx.Observable.merge(Rx.Observable.of(isSyncWmsActive(store.getState())).filter(a => a), action$.ofType(START_SYNC_WMS)).map(() => addFilterToWMSLayer(layer, filter));
+                return Rx.Observable.merge(
+                    Rx.Observable.of(isSyncWmsActive(store.getState())).filter(a => a),
+                    action$.ofType(START_SYNC_WMS))
+                    .mergeMap(() => {
+                        // if a spatial filter is present AND native crs is not present, we call the getCapabilites
+                        if (filter && filter.spatialField && filter.spatialField.geometry && filter.spatialField.geometry.coordinates && filter.spatialField.geometry.coordinates[0]) {
+                            const objLayer = getLayerFromId(store.getState(), layerId);
+                            if (!objLayer.nativeCrs) {
+                                const onlyAttributeFilter = {
+                                    ...filter,
+                                    spatialField: undefined
+                                };
+                                return Rx.Observable.of(addFilterToWMSLayer(layerId, onlyAttributeFilter));
+                            }
+                            return Rx.Observable.of(addFilterToWMSLayer(layerId, reprojectFilterInNativeCrs(filter, objLayer.nativeCrs)));
+                        }
+                        return Rx.Observable.of(addFilterToWMSLayer(layerId, filter));
+                    });
```

Comment:
I think here a takeUntil is needed to stop monitoring.
What if you stop and sync and change filter?

Reply:
it does not reach this point because in the merge function it checks if the sync is active or is being activated in that moment.